src/mpi/coll/allgather/allgather.c

```diff
@@ -101,10 +101,10 @@ int MPI_Allgather(const void *sendbuf, int sendcount, MPI_Datatype sendtype, voi
 #define MPI_Allgather PMPI_Allgather
 
 #undef FUNCNAME
-#define FUNCNAME MPIR_Allgather_intra_auto
+#define FUNCNAME MPIR_Allgather__intra__auto
```

Comment:
@AlexeyMalkhanov what is the purpose of these changes to double underscores? If this PR is to be reviewed, there needs to be separation of functional and non-functional changes into separate commits. Commit messages need to be added to explain what is going on and why.

Reply:
To be clear, it's not just functional vs. non-functional stuff. Commits should be independent and explained in the commit message. It's necessary for a decent code review.

Reply:
@ddurnov I don't think double underscores are good for function names.  The functions that are missing intra/inter are algorithms which work in both cases.  Perhaps we could use `allcomm` to suggest both intra/inter instead?

Reply:
@pavanbalaji So should we have something like: ```<MPI_Coll>_<intracomm|intercomm|allcomm>_<algorithm name>``` in that case? In the end we should have a collective operation registration infrastructure so name won't matter that much (CVAR like idea) but right now formal fixed naming scheme would be helpful. (We may fetch all available collectives from the code by using simple pattern)

Reply:
I was recommending `<MPI_Coll>_<intra|inter|allcomm>_<algorithm_name>`.

Reply:
Thanks. There was a bit of confusion on our side because of ```__``` for container structure names and collective operation ids you have added.
Would be great if we could align container names/ids with collective operation names.

Reply:
OK, if that's what people prefer, that's fine with me.  But please get some more votes on the naming before the change.

Reply:
@raffenet @wesbland @alexander-sannikov @akhillanger Do you prefer ```__``` or ```_``` in the names of containers, ids and functions?

Reply:
It doesn't make much difference to me now that the double underscore is already being used. In general, I think having the double underscore is a great way to get typos in your function names, but I guess that's what a good linter is for.

Reply:
Seems like ```_``` (single underscore) is the way